pkg/capabilities/consensus/ocr3/factory.go

```diff
@@ -37,19 +39,24 @@ func newFactory(s *requests.Store, c *capability, batchSize int, outcomePruningT
 	}, nil
 }
 
-func (o *factory) NewReportingPlugin(ctx context.Context, config ocr3types.ReportingPluginConfig) (ocr3types.ReportingPlugin[[]byte], ocr3types.ReportingPluginInfo, error) {
+func (o *factory) NewReportingPlugin(_ context.Context, config ocr3types.ReportingPluginConfig) (ocr3types.ReportingPlugin[[]byte], ocr3types.ReportingPluginInfo, error) {
+	var configProto types.ReportingPluginConfig
+	err := proto.Unmarshal(config.OffchainConfig, &configProto)
+	if err != nil {
+		return nil, ocr3types.ReportingPluginInfo{}, err
+	}
 	rp, err := newReportingPlugin(o.store, o.capability, o.batchSize, config, o.outcomePruningThreshold, o.lggr)
```

Comment:
Here o.batchSize should become configProto.maxRequestBatchSize

Reply:
You can do the same for outcomePruningThreshold - make it configurable via OCR onchain config.